CDCgov/prime-reportstream
# Engagement/jessica/15544 fhirdata api

This PR adds an API that runs the fhirdata command on a message passed to it and returns the result. It will also error out and return the errors if the filters are invalid or if they filter out everything. 

Test Steps:
1. Make sure the CLI still works. A lot of the code had to be changed, so running some basic commands through here would be ideal.
2. Make sure appropriate errors are returned when a receiver isn't found, a filter is broken, or everything is filtered out.

## Changes
- Adds an API that runs the fhirdata command on a message. 
- Adds some tests to the fhirdata stuff. 

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
